staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -244,6 +247,55 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		go finalizingController.Run(5, context.StopCh)
 		return nil
 	})
+	s.GenericAPIServer.AddPostStartHookOrDie("crd-discovery-available", func(context genericapiserver.PostStartHookContext) error {
+		return wait.PollImmediateUntil(100*time.Millisecond, func() (bool, error) {
+			// only check if we have a valid list for a given resourceversion
+			if !s.Informers.Apiextensions().InternalVersion().CustomResourceDefinitions().Informer().HasSynced() {
```

Comment:
just for the record, the usage of internal informer here will be conflicting w/ the effort from https://github.com/kubernetes/kubernetes/pull/84005. use GA CRD would be better.

Reply:
That's fair, lemme update that right away

Reply:
Honestly, it looks like either of us will have to rebase on top of the other one. I don't want to introduce double logic for obtaining clients temporarily.

Reply:
it looks like yours is ready and since it's bigger it'll be easier for me to rebase. I'll wait.